ql/src/test/org/apache/hadoop/hive/ql/io/arrow/TestSerializer.java

```diff
@@ -0,0 +1,72 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.io.arrow;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class TestSerializer {
+  @Test
+  public void testEmptyArray() {
+    List<TypeInfo> typeInfos = TypeInfoUtils.getTypeInfosFromTypeString("array<tinyint>");
+    List<String> fieldNames = Arrays.asList(new String[]{"a"});
+    Serializer converter = new Serializer(new HiveConf(), "attemptId", typeInfos, fieldNames);
+    ArrowWrapperWritable writable = converter.emptyBatch();
+    Assert.assertEquals("Schema<a: List<$data$: Int(8, true)>>",
+        writable.getVectorSchemaRoot().getSchema().toString());
+  }
+
+  @Test
+  public void testEmptyStruct() {
+    List<TypeInfo> typeInfos = TypeInfoUtils.getTypeInfosFromTypeString("struct<b:int,c:string>");
+    List<String> fieldNames = Arrays.asList(new String[] { "a" });
+    Serializer converter = new Serializer(new HiveConf(), "attemptId", typeInfos, fieldNames);
+    ArrowWrapperWritable writable = converter.emptyBatch();
+    Assert.assertEquals("Schema<a: Struct<b: Int(32, true), c: Utf8>>",
+        writable.getVectorSchemaRoot().getSchema().toString());
+  }
+
+  @Test
+  public void testEmptyMap() {
+    List<TypeInfo> typeInfos = TypeInfoUtils.getTypeInfosFromTypeString("map<string,string>");
+    List<String> fieldNames = Arrays.asList(new String[] { "a" });
+    Serializer converter = new Serializer(new HiveConf(), "attemptId", typeInfos, fieldNames);
+    ArrowWrapperWritable writable = converter.emptyBatch();
+    Assert.assertEquals("Schema<a: List<$data$: Struct<keys: Utf8, values: Utf8>>>",
+        writable.getVectorSchemaRoot().getSchema().toString());
+  }
+
+  @Test
+  public void testEmptyComplexStruct() {
+    List<TypeInfo> typeInfos = TypeInfoUtils.getTypeInfosFromTypeString(
+        "struct<b:array<tinyint>,c:map<string,string>,d:struct<e:array<tinyint>,f:map<string,string>>>");
+    List<String> fieldNames = Arrays.asList(new String[] { "a" });
+    Serializer converter = new Serializer(new HiveConf(), "attemptId", typeInfos, fieldNames);
+    ArrowWrapperWritable writable = converter.emptyBatch();
+    Assert.assertEquals(
+        "Schema<a: Struct<b: List<$data$: Int(8, true)>, c: List<$data$: Struct<keys: Utf8, values: Utf8>>, " +
+            "d: Struct<e: List<$data$: Int(8, true)>, f: List<$data$: Struct<keys: Utf8, values: Utf8>>>>>",
+        writable.getVectorSchemaRoot().getSchema().toString());
+  }
```

Comment:
Would it make sense to test nested types here as well? like list<list> ?

Reply:
Addes some more tests to cover every nested type at least once